apm-sniffer/apm-sdk-plugin/mysql-common/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/PreparedStatementExecuteMethodsInterceptor.java

```diff
@@ -78,4 +108,39 @@ public final Object afterMethod(EnhancedInstance objInst, Method method, Object[
     private String buildOperationName(ConnectionInfo connectionInfo, String methodName, String statementName) {
         return connectionInfo.getDBType() + "/JDBI/" + statementName + "/" + methodName;
     }
+
+    private Object getDisplayedParameter(final Method method, final Class<?>[] argumentTypes, final Object[] allArguments) {
+        final String methodName = method.getName();
+        if ("setNull".equals(methodName)) {
+            return "null";
+        }
+        if ("setObject".equals(methodName)) {
+            final Object parameter = allArguments[0];
+            final Class<?> parameterType = argumentTypes[1];
+
+            if (DISPLAYABLE_TYPES.contains(parameterType)) {
+                return parameter;
+            }
+        } else if (PS_SETTERS.contains(methodName)) {
+            return allArguments[1];
+        }
+        return SQL_PARAMETER_PLACEHOLDER;
+    }
+
+    private String buildParameterString(Object[] parameters) {
```

Comment:
Why don't use GSON Array `#toString`?

Reply:
We cannot use Gson because the target application don't necessary include it

Reply:
Yes, ignore this.